controllers/chat/group/removeGroupMember.js

```diff
@@ -26,6 +26,31 @@ const removeGroupMember = async (req, res) => {
 
     let all_members = queriedChannel[0].data.better_channel_member.map((member) => member.user.id);
 
+    const textOwnUser = `You removed ${targetUserModel.username} from this group`;
+    const textTargetUser = `${ownUser.username} removed you from this group`;
+    const textDefaultUser = `${ownUser.username} removed ${targetUserModel.username} from this group`;
+
+    await currentChannel.sendMessage(
+      {
+        text: textDefaultUser,
+        own_text: textOwnUser,
+        other_text: textTargetUser,
+        other_system_user: targetUserModel.user_id,
+        better_type: 'remove_member_from_group',
+        type: 'system',
+        user_id: userId,
+        only_to_user_show: userId,
+        disable_to_user: false,
+        is_from_prepopulated: true,
+        system_user: userId,
+        isSystem: true,
+        members: all_members
+      },
+      {
+        skip_push: true
+      }
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change in the sequence of operations seems logical. Sending a message before removing a user ensures that the user receives the notification about their removal. However, there should be error handling around `currentChannel.sendMessage` and `Getstream.chat.removeGroupMember`. If either operation fails, the other should not proceed to maintain consistency.

```diff
+    try {
+      await currentChannel.sendMessage(
+        {
+          text: textDefaultUser,
+          own_text: textOwnUser,
+          other_text: textTargetUser,
+          other_system_user: targetUserModel.user_id,
+          better_type: 'remove_member_from_group',
+          type: 'system',
+          user_id: userId,
+          only_to_user_show: userId,
+          disable_to_user: false,
+          is_from_prepopulated: true,
+          system_user: userId,
+          isSystem: true,
+          members: all_members
+        },
+        {
+          skip_push: true
+        }
+      );
+    } catch (error) {
+      console.error('Failed to send message:', error);
+      throw error;
+    }

+    let response;
+    try {
+      response = await Getstream.chat.removeGroupMember(channelId, userId, targetUserId);
+    } catch (error) {
+      console.error('Failed to remove group member:', error);
+      throw error;
+    }
+    const {channel, channelApiResponse} = response.data || {};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->